### PR TITLE
Add format weighting ab test

### PR DESCRIPTION
**Pull request by brenetic**
We are going to be running an A/B test to test changes to format
weighting.

Merge after https://github.com/alphagov/finder-frontend/pull/318

Paired with @suzannehamilton 

https://trello.com/c/JJ3mIOP1

### CODE REVIEW

**Comment by suzannehamilton** [comment]
@MatMoore: That makes sense. I've updated the names. 👍

**Comment by suzannehamilton** [comment]
I've just moved the A/B meta tag from `<body>` to `<head>`. The tests will fail until #318 is merged, because it fixes the header rendering.

File: app/controllers/search_controller.rb [hunk 1]
```diff
@@ -24,7 +29,9 @@ def index
     if search_params.no_search? && params[:format] != "json"
       render(action: 'no_search_term') && return
     end
-    search_response = SearchAPI.new(search_params).search
+
+    variant = search_ab_test_variant.variant_name
+    search_response = SearchAPI.new(search_params, format_boosting: variant).search
```
Comment (dwhenry) [inline app/controllers/search_controller.rb]: we have called the variant `format_boosting` and the test `FormatWeighting` can we get these names to match?

File: spec/controllers/search_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        "suggested_queries": []
+      }|
+
+    stub_request(:get, /search.json/).to_return(status: 200, body: rummager_response, headers: {})
+  end
+
+  context "Format weighting A/B test" do
+    it "should set the correct A variant tags" do
+      with_variant FormatWeighting: "A" do
+        get :index, params: { q: "cheese" }
```
Comment (dwhenry) [inline spec/controllers/search_controller_spec.rb]: This test doesn't appear to have a test? or at least not one that relates to the test name `should set the correct A variant tags`

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -17,6 +17,10 @@
 # If you are not using ActiveRecord, you can remove this line.
 ActiveRecord::Migration.check_pending! if defined?(ActiveRecord::Migration)
 
+GovukAbTesting.configure do |config|
+  config.acceptance_test_framework = :active_support
+end
```
Comment (dwhenry) [inline spec/spec_helper.rb]: this might make more sense in a `spec/support/govuk_ab_testing.rb` file rather than here.

**Comment by dwhenry** [comment]
Also, the description doesn't reflect the code which is confusing and we should mention the dependancy on https://github.com/alphagov/finder-frontend/pull/318/files being merged

File: spec/controllers/search_controller_spec.rb (hunk 2 above)
Comment (brenetic) [reply spec/controllers/search_controller_spec.rb]: `with_variant` asserts meta tags have been set correctly amongst other things

**Pull request merged**